File: python/src/node_v2.py
#!/usr/bin/env python3
# laurent.martin.aspera@fr.ibm.com
# Upload files using node API and transfer spec v2
import utils.tools
import utils.transfer_client

test_env = utils.tools.Tools()
transfer_client = utils.transfer_client.TransferClient(test_env).setup()

try:
    # get node information from config file
    config = test_env.conf('node')

    # prepare transfer spec v2 for COS
    t_spec = {
        'title': 'send using Node API and ts v2',
        'session_initiation': {
            'node_api': {
                'url': config['url'],
                'headers': [
                    utils.tools.basic_auth_header_key_value(config['user'], config['pass'])
                ]
            }
        },
        'direction': 'send',
        'assets': {
            'destination_root': config['folder_upload'],
            'paths': []
        },
    }

    # add file list in transfer spec
    for f in test_env.file_list():
        t_spec['assets']['paths'].append({'source': f})

    # start transfer, using Transfer SDK
    transfer_client.start_transfer_and_wait(t_spec)
finally:
    transfer_client.shutdown()

File: cpp/src/node_v2.cpp
#include "utils/rest.hpp"
#include "utils/test_environment.hpp"

int main(const int argc, const char* const argv[]) {
    try {
        utils::TestEnvironment test_env(argc, argv);
        test_env.startup();
        const std::string node_api_url = test_env.conf_str({"node", "url"});
        const std::string header_authorization = utils::Rest::basic_auth_header(test_env.conf_str({"node", "user"}), test_env.conf_str({"node", "pass"}));
        json::object transfer_spec_v2 = {
            {"title", "send using Node API and ts v2"},
            {"session_initiation",
             {{"node_api",
               {{"url", node_api_url},
                {"headers",
                 json::array{
                     {{"key", "Authorization"},
                      {"value", header_authorization}}}}}}}},
            {"direction", "send"},
            {"assets",
             {{"destination_root", test_env.conf_str({"node", "folder_upload"})},
              {"paths", json::array()}}}};
        test_env.add_files_to_ts(transfer_spec_v2["assets"].as_object()["paths"].as_array());
        test_env.start_transfer_and_wait(transfer_spec_v2);
        return 0;
    } catch (const std::exception& e) {
        std::clog << "Exception: " << e.what() << std::endl;
        return 1;
    }
}
